### PR TITLE
Add SQLite support

### PR DESCRIPTION
**Pull request by Fokko**
image

### CODE REVIEW

**Comment by Fokko** [comment]
@jayceslesar Thanks for chiming in here! If I'm not mistaken, PyIceberg could serve as a backend behind the Ibis front end. We currently use sqlalchemy as an ORM for different database implementations.

File: mkdocs/docs/configuration.md [hunk 1]
```diff
@@ -150,6 +151,22 @@ catalog:
     uri: postgresql+psycopg2://username:password@localhost/mydatabase
 ```
 
+In the case of SQLite:
+
+<!-- prettier-ignore-start -->
+
+!!! warning inline end "Development only"
+    SQLite is not built for concurrency, and should only be used for exploratory or development purposes.
```
Comment (bitsondatadev) [inline mkdocs/docs/configuration.md:159]: Avoid passive.

```suggestion
    SQLite is not built for concurrency, you should use this catalog for exploratory or development purposes.
```

File: tests/catalog/test_sql.py [hunk 2]
```diff
@@ -21,6 +21,7 @@
 
 import pytest
 from pytest import TempPathFactory
+from pytest_lazyfixture import lazy_fixture
```
Comment (bitsondatadev) [inline tests/catalog/test_sql.py:24]: Clever :)

**Review by bitsondatadev — APPROVED** [review]
Few nits in the docs, but LGTM!

**Comment by Fokko** [comment]
Thanks for the suggestions @bitsondatadev 🙌

**Review by HonahX — APPROVED** [review]

**Review by rdblue — APPROVED** [review]

**Pull request merged**